device_service.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"errors"
+	"log"
+)
+
+const (
+	deviceCreated = iota
+	validationError
+	savingError
+)
+
+type Device struct {
+	Id       int     `json:"id"`
+	Name     string  `json:"name"`
+	Interval int     `json:"interval"`
+	Value    float32 `json:"value"`
+}
+
+type deviceDAO interface {
+	Save(device Device) (Device, error)
+}
+
+type DeviceService struct {
+	dao deviceDAO
+}
+
+func (s *DeviceService) CreateDevice(device Device) (int, Device, error) {
```

Comment:
This return signature is not obvious. You should either return an error or and int. But in golang the natural way is to return an error.

I think you have two options:
1. you can use `errors.New` and have some error constants so that you can "if" on it in the handler
2. create custom error; for example:

```golang
type VeryGenericError string

func (v VeryGenericError) Error() String {
   return string(v)
}

const (
  ErrorInvalidPayload = VeryGenericError("invalid payload")
)
```